recipes/sundials/all/test_package/test_package_v6.c

```diff
@@ -0,0 +1,18 @@
+#include <nvector/nvector_serial.h>
```

Comment:
we need to document why do we need different code for the test pacakge depending on the version. At least we should provide a hint in the description of this PR. This might be obvious, but I am seeing this from the eyes of a complete outsider to the library itself. Thanks!

Reply:
So I should add something like this as a comment, I guess?
> SUNDIALS v6.0.0 introduced a new SUNContext object on which all other SUNDIALS objects depend.
https://github.com/LLNL/sundials/blob/main/CHANGELOG.md#changes-to-sundials-in-release-600

Reply:
Done.